### PR TITLE
DAOS-6903 Test: Fix regression in NVMe health test code.

### PR DESCRIPTION
**Pull request by ravalsam**
Quick-Functional: true
Test-tag: nvme_health

Signed-off-by: Samir Raval <samir.raval@intel.com>

### CODE REVIEW

**Comment by ravalsam** [comment]
Three test case failure are because of DAOS-5922

**Comment by ravalsam** [comment]
Recent Two test failure are because of DAOS-6932

**Review by rpadma2 — DISMISSED** [review]
LGTM

**Comment by ravalsam** [comment]
enospace.py and nvme_health.py Ci result can be found from 
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-4832/3/artifact/Functional_Hardware_Medium/nvme/

File: src/tests/ftest/nvme/nvme_health.py [hunk 1]
```diff
@@ -30,21 +30,18 @@ def test_monitor_for_large_pools(self):
...
-        #Stop the servers to run SPDK too to get the server capacity
-        self.stop_servers()
-        storage = self.get_nvme_max_capacity()
-        self.start_servers()
+        storage = self.get_max_storage_sizes()
 
-        #Create the pool from 80% of available of storage space
-        single_pool_nvme_size = int((storage * 0.80)/no_of_pools)
+        #Create the pool from 90% of available of storage space
+        single_pool_nvme_size = int((storage[1] * 0.90)/no_of_pools)
```
Comment (rpadma2) [inline src/tests/ftest/nvme/nvme_health.py:36]: Just a thought... Can we move the percentage to YAML file parameter? In this way, you get modify in the YAML for future without need to update the script.
eg:
fill_max_pool_capacity : 90

File: src/tests/ftest/nvme/nvme_health.py [hunk 2]
```diff
@@ -30,21 +30,18 @@ def test_monitor_for_large_pools(self):
...
+        single_pool_scm_size = int((storage[0] * 0.90)/no_of_pools)
```
Comment (rpadma2) [inline src/tests/ftest/nvme/nvme_health.py:37]: same as above.

File: src/tests/ftest/nvme/nvme_health.py (hunk 1 above)
Comment (ravalsam) [reply src/tests/ftest/nvme/nvme_health.py:36]: It's not major as both the cases it has to change, but we can update later as I do not want to restart the full Ci pipeline which will take another day to finish.

Comment (rpadma2) [reply src/tests/ftest/nvme/nvme_health.py:36]: Sure. No problem with me...

**Review by rpadma2 — DISMISSED** [review]

File: src/tests/ftest/nvme/nvme_health.py (hunk 1 above)
Comment (rpadma2) [reply src/tests/ftest/nvme/nvme_health.py:36]: With all tests passed. I think we should table the changes later... I didn't see that CI has completed.

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Comment by ravalsam** [comment]
Latest result of ENOSPACE and NVMe health tests from Ci https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-4832/31/artifact/Functional_Hardware_Medium/nvme/

File: src/tests/ftest/nvme/nvme_health.py (hunk 1 above)
Comment (ravalsam) [reply src/tests/ftest/nvme/nvme_health.py]: Updated the test and added fill percentage in yaml.

File: src/tests/ftest/nvme/nvme_health.py (hunk 2 above)
Comment (ravalsam) [reply src/tests/ftest/nvme/nvme_health.py]: Updated the test and added fill percentage in yaml.

**Review by saurabhtandan — COMMENTED** [review]
Code wise LGTM, but is there a successful run for nvme_io.py test?  Last run for this test I seen for build number 30 and it failed.

**Comment by ravalsam** [comment]
> Code wise LGTM, but is there a successful run for nvme_io.py test? Last run for this test I seen for build number 30 and it failed.

No, it's not there are other issues which I am working on for nvme_io.py but the changes I made in this PR are needed as well. Once I fixed those issue will have to create another PR.

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Comment by saurabhtandan** [comment]
If a separate ticket is needed for nvme_io can we please create that, if it is not being worked upon as part of same jira ticket.

**Review by dinghwah — APPROVED** [review]
LGTM

**Comment by ravalsam** [comment]
> If a separate ticket is needed for nvme_io can we please create that, if it is not being worked upon as part of same jira ticket.

Yes, will create one.

**Pull request merged**